app/serializers/discussion_reader_serializer.rb

```diff
@@ -9,5 +10,13 @@ class DiscussionReaderSerializer < ActiveModel::Serializer
              :read_items_count,
              :read_salient_items_count,
              :last_read_sequence_id,
-             :volume
+             :volume,
+             :is_read
+  def is_read
+    object.last_read_sequence_id == object.discussion.last_sequence_id
+  end
+
+  def group_id
+    object.discussion.group_id
```

Comment:
I don't think it should be a stored value, otherwise it will wind up being wrong. How about you do a loki where() and pass a function to filter unread. if you do it last it should not be very expensive.